lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -415,6 +417,22 @@ def add_payment_method(xml, amount, payment_method, options)
         end
       end
 
+      def add_network_tokenization_card(xml, payment_method)
+        xml.paymentDetails do
+          xml.tag! 'EMVCO_TOKEN-SSL', 'type' => 'NETWORKTOKEN' do
+            xml.tokenNumber payment_method.number
+            xml.expiryDate do
+              xml.date(
+                'month' => format(payment_method.month, :two_digits),
+                'year' => format(payment_method.year, :four_digits)
+              )
+            end
+            xml.cryptogram payment_method.payment_cryptogram
+            xml.eciIndicator payment_method.eci
```

Comment:
Will this result in an empty tag In case `eci` is not set in the payment method. If so, I'd double check if Worldpay is okay with an empty tag or prefers a default value or no tag when eci is not available.

Reply:
I tested against the sandbox with an empty tag and there were no issues

Reply:
From https://developer.worldpay.com/docs/wpg/reference/decryptwallets

>Zero padded to 2 digit format (if a value of 7 is received, then add a value of 07 to the XML)

@Klaiv  Might have to format the value like is done for month and year.

Reply:
@shasum Thank you for catching that, this is resolved now in the latest commit